sdk/src/suins-client.ts

```diff
@@ -0,0 +1,165 @@
+
+		if (network === 'mainnet') {
+			this.constants = MAINNET_CONFIG;
+		}
+
+		if (network === 'testnet') {
+			this.constants = TESTNET_CONFIG;
+		}
+
+		if (config.packageIds) {
+			this.constants = { ...(this.constants || {}), ...config.packageIds };
+		}
+	}
+
+	/**
+	 * Returns the price list for SuiNS names.
+	 */
+	async getPriceList(): Promise<SuinsPriceList> {
+		if (!this.constants.suinsObjectId) throw new Error('Suins object ID is not set');
+		if (!this.constants.suinsPackageId) throw new Error('Price list config not found');
+
+		const priceList = await this.#client.getDynamicFieldObject({
+			parentId: this.constants.suinsObjectId,
+			name: {
+				type: getConfigType(
+					this.constants.suinsPackageId.v1,
+					getPricelistConfigType(this.constants.suinsPackageId.v1),
+				),
+				value: { dummy_field: false },
+			},
+		});
+
+		if (
+			!priceList ||
+			!priceList.data ||
+			!priceList.data.content ||
+			priceList.data.content.dataType !== 'moveObject' ||
+			!('value' in priceList.data.content.fields)
+		)
+			throw new Error('Price list not found');
+
+		const contents = priceList.data.content.fields.value as Record<string, any>;
+
+		return parsePriceListFromConfig(contents);
+	}
+
+	async getRenewalPriceList(): Promise<SuinsPriceList> {
+		if (!this.constants.suinsObjectId) throw new Error('Suins object ID is not set');
+		if (!this.constants.suinsPackageId) throw new Error('Price list config not found');
+		if (!this.constants.renewalPackageId) throw new Error('Renewal package ID is not set');
+
+		const priceList = await this.#client.getDynamicFieldObject({
+			parentId: this.constants.suinsObjectId,
+			name: {
+				type: getConfigType(
+					this.constants.suinsPackageId.v1,
+					getRenewalPricelistConfigType(this.constants.renewalPackageId),
+				),
+				value: { dummy_field: false },
+			},
+		});
+
+		if (
+			!priceList ||
+			!priceList.data ||
+			!priceList.data.content ||
+			priceList.data.content.dataType !== 'moveObject' ||
+			!('value' in priceList.data.content.fields)
+		)
+			throw new Error('Price list not found');
+
+		const contents = (priceList.data.content.fields.value as Record<string, any>)?.fields?.config;
+
+		return parsePriceListFromConfig(contents);
+	}
+
+	async getNameRecord(name: string): Promise<NameRecord> {
+		if (!isValidSuiNSName(name)) throw new Error('Invalid SuiNS name');
+		if (!this.constants.suinsPackageId) throw new Error('Suins package ID is not set');
+		if (!this.constants.registryTableId) throw new Error('Registry table ID is not set');
+
+		const nameRecord = await this.#client.getDynamicFieldObject({
+			parentId: this.constants.registryTableId,
+			name: {
+				type: getDomainType(this.constants.suinsPackageId.v1),
+				value: name.split('.').reverse(),
+			},
+		});
+		const fields = nameRecord.data?.content;
+
+		if (nameRecord.error || !fields || fields.dataType !== 'moveObject')
+			throw new Error('Name record not found. This domain is not registered.');
```

Comment:
I'd recommend adding prettier (and maybe eslint) to this repo

Reply:
Oh yes, I got used to having prettier/eslint by default on the monorepo where I last did changes.